### PR TITLE
Update design to allow custom attributes in call to Notification service

### PR DESCRIPTION
**Pull request by sbose78**
Added support for passing a `custom` parameter to the `notify` API.
This would be used to pass the verifyCode for an updated email.

Part of https://github.com/fabric8-services/fabric8-auth/issues/62

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ func main() {
 	resolvers.Register("workitem.update", collector.ConfiguredVars(config, collector.NewWorkItemResolver(witClient)))
 	resolvers.Register("comment.create", collector.ConfiguredVars(config, collector.NewCommentResolver(witClient)))
 	resolvers.Register("comment.update", collector.ConfiguredVars(config, collector.NewCommentResolver(witClient)))
+	resolvers.Register("user.email.update", collector.ConfiguredVars(config, collector.NewUserResolver(witClient)))
```
Comment (aslakknutsen) [inline main.go]: Missing template ?

**Comment by sbose78** [comment]
Used the workitem.update template to have a super simple email verification template till have something nicer done ( by myself or UX :) )

screenshot from 2017-12-18 20-33-50

File: main.go (hunk 1 above)
Comment (sbose78) [reply main.go]: added a fork of workitem.update template where I stripped out almost everything. Will make it look pretty  https://github.com/fabric8-services/fabric8-notification/pull/51/commits/8f420859f8488cf74226e5c2171becb511f8b671

File: email/notifier.go [hunk 2]
```diff
@@ -80,6 +81,13 @@ func (a *AsyncWorkerNotifier) do(cn contextualNotification) {
 
 		return
 	}
+
+	if vars == nil {
```
Comment (aslakknutsen) [inline email/notifier.go]: Can't assume vars is null here..

File: template/user.email.update/body.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      </div>
+    </div>
+    <div style="
+          background-color: #f9f9f9;
+          padding: 10px 20px;
+          min-height: 150px;
+          max-width: 700px;
+          text-align: left;">
+      <h3 style="margin-bottom: 5px;">
+        <a href="{{.custom.VerifyURL }}" style="
```
Comment (aslakknutsen) [inline template/user.email.update/body.html]: Normal would be a camelcase style variable..

File: template/user.email.update/headers.prop [hunk 4]
```diff
@@ -0,0 +1,2 @@
+In-Reply-To: <workitems/{{.workitem.Data.ID}}@notify.openshift.io>
```
Comment (aslakknutsen) [inline template/user.email.update/headers.prop]: No workitem in the auth context. None of these headers make any sense.. ?

File: email/notifier.go (hunk 2 above)
Comment (sbose78) [reply email/notifier.go]: right, my bad..

**Comment by sbose78** [comment]
@aslakknutsen , thank you for the review - I've addressed the comments.

File: template/user.email.update/headers.prop (hunk 4 above)
Comment (sbose78) [reply template/user.email.update/headers.prop]: removed it :)

File: template/user.email.update/body.html (hunk 3 above)
Comment (sbose78) [reply template/user.email.update/body.html]: https://github.com/fabric8-services/fabric8-notification/pull/51/commits/071502685301a39eee6a9f2edd45e5466b720dbb

**Comment by sbose78** [comment]
Thanks @aslakknutsen 

added some more tests 
- https://github.com/fabric8-services/fabric8-notification/pull/51/commits/4cf85419128978c8e07a8eaeae23becef78bbae6
- https://github.com/fabric8-services/fabric8-notification/pull/51/commits/a3cdd09d241993d696f5d97ff7419255d77ea9bc

**Comment by sbose78** [comment]
Added a validator of the optional `custom` param.
https://github.com/fabric8-services/fabric8-notification/pull/51/commits/3d55bbf6de2807ab5c27c2f2c516615fac1d8dda

**Pull request merged**